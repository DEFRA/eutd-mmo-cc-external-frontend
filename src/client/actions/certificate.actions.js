import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};