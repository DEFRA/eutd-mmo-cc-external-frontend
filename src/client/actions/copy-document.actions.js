import { showFullPageError, apiCallFailed, isError, CLEAR_ERRORS } from './index';
export const CONFIRM_COPY_DOCUMENT            = 'confirm_copy_document';
export const UNAUTHORISED_COPY_DOCUMENT       = 'unauthorised_copy_document';
export const CLEAR_COPY_DOCUMENT              = 'clear_copy_document';
export const REMOVE_CONFIRM_COPY_CERTIFICATE  = 'remove_confirm_copy_certificate';
export const ADD_ERROR_COPY_CERTIFICATE       = 'add_error_copy_certificate';
export const REMOVE_ERROR_COPY_CERTIFICATE    = 'remove_error_copy_certificate';

export const submitCopyCertificate = (payload) => async (dispatch, _getState, {orchestrationApi}) => {
  const {documentNumber, copyDocumentAcknowledged, voidOriginal, excludeLandings, journey} = payload;
  const config = {headers: {documentnumber: documentNumber}};

  try {
    let ipAddress;
    if (voidOriginal) {
      const ip = await orchestrationApi.get('/client-ip');
      ipAddress = ip.data;
    }

    return await orchestrationApi.post('/confirm-copy-certificate', { copyDocumentAcknowledged, voidOriginal, excludeLandings, journey, ipAddress }, config)
    .then(res => {
      dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: {
        documentNumber: res.data.documentNumber,
        copyDocumentAcknowledged: res.data.copyDocumentAcknowledged,
        voidDocumentConfirm: res.data.voidOriginal
      }});
      return res.data.newDocumentNumber;
    });
  } catch (e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch({ type: UNAUTHORISED_COPY_DOCUMENT }); return;
    } else {
      dispatch(apiCallFailed(e.response));
    }
  }
};

export const checkCopyCertificate = (documentNumber) => async (dispatch, _getState, {orchestrationApi}) => {
  const config = {
    headers: {
      documentnumber: documentNumber
    }
  };

  try {
    const res = await orchestrationApi.get('/check-copy-certificate', config);
    const { canCopy } = res.data;
    if (!canCopy) {
      dispatch(unauthorised());
    }
  } catch(e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch(unauthorised());
      throw new Error(`Unauthorised access ${e.response.status}`);
    } else {
      dispatch(apiCallFailed(e.response));
      throw new Error(`An error has occurred ${e.response.status} (${e.response.statusText})`);
    }
  }

  function unauthorised() { return { type: UNAUTHORISED_COPY_DOCUMENT }; }
};

export const addCopyDocument = (data) => (dispatch) => dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: data });
export const clearErrors = () => (dispatch) =>  dispatch({ type: CLEAR_ERRORS });
export const clearCopyDocument = () => (dispatch) => dispatch({ type: CLEAR_COPY_DOCUMENT });
export const unauthorisedCopyDocument = () => (dispatch) => dispatch({ type: UNAUTHORISED_COPY_DOCUMENT });