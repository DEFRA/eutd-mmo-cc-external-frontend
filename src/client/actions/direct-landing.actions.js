import { beginApiCall, apiCallFailed, isForbidden, showFullPageError, CLEAR_ERRORS, isSilverLineError } from './index';
import { exportPayloadActionTypes } from '../actions';


export const GET_DIRECT_LANDING_PRODUCTS = 'get_direct_landing_products';
export const CLEAR_DIRECT_LANDING = 'clear_direct_landing';
export const DIRECT_LANDING_PRODUCTS_FAILED = 'direct_landing_products_failed';
export const POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'post_direct_landing_products_unauthorised';
export const GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'get_direct_landing_products_unauthorised';
export const POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR = 'post_direct_landing_prodcts_waf_error';

export const getDirectLandings =
  (documentNumber) =>
  async (dispatch, _getState, { orchestrationApi }) => {
    const config = { headers: { documentnumber: documentNumber } };

    try {
      dispatch(beginApiCall());

      const res = await orchestrationApi.get(
        '/export-certificates/export-payload/direct-landings',
        config
      );

      await dispatch({ type: GET_DIRECT_LANDING_PRODUCTS, payload: res.data });
    } catch (e) {
      if (isForbidden(e)) {
        dispatch({ type: GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
      } else {
        dispatch(apiCallFailed(e.response));
        throw new Error(
          `An error has occurred ${e.response.status} (${e.response.statusText})`
        );
      }
    }
  };

export const clearDirectLanding = () => ({ type: CLEAR_DIRECT_LANDING });

export const upsertDirectLanding = (landing, documentNumber = undefined) => (dispatch, getState, { orchestrationApi } ) => {
  const config = (documentNumber)
    ? {
        headers: {
          documentnumber: documentNumber
        }
    }
    : {};

  return orchestrationApi.post('/export-certificates/direct-landing/validate', landing, config)
    .then(
      res => {
        dispatch(landingUpserted(res.data));
        dispatch(clearErrors());
      },
      error => {
        if (error.response.status === 400) {
          dispatch(landingUpserted(error.response.data));
          throw new Error();
        } else if (isSilverLineError(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR, supportID: error.response.data.match(/\d+/)[0] });
        } else if (isForbidden(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
        } else {
          dispatch(showFullPageError());
        }
      });

  function landingUpserted(exportPayload) { return { type: exportPayloadActionTypes.LANDING_UPSERTED, exportPayload, errors: exportPayload.errors }; }
  function clearErrors() { return { type: CLEAR_ERRORS }; }
};
