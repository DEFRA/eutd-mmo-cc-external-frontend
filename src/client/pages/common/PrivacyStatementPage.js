import * as React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Redirect } from 'react-router-dom';

import { saveUserAttribute, getAllUserAttributes } from '../../actions/userAttributes.actions';
import PrivacyStatementEnglish from './PrivacyStatementEnglish';
import PrivacyStatementWelsh from './PrivacyStatementWelsh';

const PRIVACY_STATEMENT_ATTRIBUTE_NAME = 'privacy_statement';

class PrivacyStatementPage extends React.Component {
  constructor(props) {
    super(props);
  }

  onSubmit = async (e) => {
    e.preventDefault();
    // If it's submitted then consider it they agreed
    const payload = {
      key: PRIVACY_STATEMENT_ATTRIBUTE_NAME,
      value: true
    };
    await this.props.saveUserAttribute(payload);
    this.props.route.alwaysShowPrivacyPage &&
      this.props.history.push('/');
  }

  handleBackLinkClick = (e) => {
    e.preventDefault();

    this.props.history.goBack();
  };

  async componentDidMount() {
    await this.props.getAllUserAttributes();
  }

  getSelectedLanguage = (userAttributes) => {
    const userLanguageAttribute = userAttributes.find(
      (userAttribute) => userAttribute.name === 'language'
    );
    if (userLanguageAttribute) {
      return userLanguageAttribute.value;
    }
  };

  showPrivacyNotice = () => {
    const { nextUri } = this.props.route;
    const { userAttributes } = this.props;
    const selectedLanguage = this.getSelectedLanguage(userAttributes);
    const privacyAcceptedAlready = this.privacyAcceptedAlready();

    return selectedLanguage === 'cy_UK' ? (
      <PrivacyStatementWelsh
        onSubmit={this.onSubmit}
        privacyAcceptedAlready={privacyAcceptedAlready}
        nextUri={nextUri}
        onBackLinkClick={this.handleBackLinkClick}
      />
     ) : (
      <PrivacyStatementEnglish
        onSubmit={this.onSubmit}
        privacyAcceptedAlready={privacyAcceptedAlready}
        nextUri={nextUri}
        onBackLinkClick={this.handleBackLinkClick}
      />
    );
  };

  redirectToDashboard = () => {
    const { nextUri } = this.props.route;
    return <Redirect to={nextUri} />;
  };

  privacyAcceptedAlready = () => {
    let acceptedAlready = false;
    const { userAttributes } = this.props;
    for (let attribute of userAttributes) {
      if (attribute.name === PRIVACY_STATEMENT_ATTRIBUTE_NAME) {
        acceptedAlready = true;
        break;
      }
    }
    return acceptedAlready;
  };

  render = () => {
    const privacyAcceptedAlready = this.privacyAcceptedAlready();
    const { alwaysShowPrivacyPage } = this.props.route;
    if (alwaysShowPrivacyPage) {
      return this.showPrivacyNotice();
    } else {
      if (privacyAcceptedAlready) {
        return this.redirectToDashboard();
      } else {
        return this.showPrivacyNotice();
      }
    }
  };
}

const mapStateToProps = state => {
  return {
    userAttributes: state.userAttributes
  };
};

const loadData = store => {
  return store.dispatch(getAllUserAttributes());
};

export default {
  component: withRouter(connect(mapStateToProps, {
    saveUserAttribute,
    getAllUserAttributes
  })(PrivacyStatementPage)),
  loadData
};