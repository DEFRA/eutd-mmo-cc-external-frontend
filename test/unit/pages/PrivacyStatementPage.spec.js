import { mount } from 'enzyme';
import * as React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import * as PrivacyStatementPage from '../../../src/client/pages/common/PrivacyStatementPage';
import PrivacyStatementPageWrapper from '../../../src/client/pages/common/PrivacyStatementPage';
import { getAllUserAttributes } from '../../../src/client/actions/userAttributes.actions';

jest.mock('../../../src/client/actions/userAttributes.actions');

describe('Privacy page', () => {
  describe('when user has not Accepted privacy', () => {
    let wrapper;
    const mockStore = configureStore([thunk]);
    const mockPush = jest.fn();

    const getWrapper = (selectedLanguage) => {
      const store = mockStore({
        userAttributes: [
          {
            name: 'language',
            value: selectedLanguage,
          },
        ],
      });

      const props = {
        route: {
          alwaysShowPrivacyPage: true,
        },
        history: mockPush,
        saveUserAttribute: jest.fn()
      };

      window.scrollTo = jest.fn();

      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <PrivacyStatementPage.default.component {...props} />
          </MemoryRouter>
        </Provider>
      );

      return wrapper;
    };

    it('should render the page in English if no language has been selected', () => {
      const store = mockStore({
        userAttributes: [],
      });

      const props = {
        route: {},
      };

      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <PrivacyStatementPage.default.component {...props} />
          </MemoryRouter>
        </Provider>
      );

      expect(wrapper).toBeDefined();
      expect(
        wrapper.find('PageTitle[title=\'Privacy - GOV.UK\']').exists()
      ).toBeTruthy();
    });

    it('should render the page successfully when the selected language is English', () => {
      const wrapper = getWrapper('en_UK');
      expect(wrapper).toBeDefined();
      expect(
        wrapper.find('PageTitle[title=\'Privacy - GOV.UK\']').exists()
      ).toBeTruthy();
    });

    it('should render the page successfully when the selected language is Welsh', () => {
      const wrapper = getWrapper('cy_UK');
      expect(wrapper).toBeDefined();
      expect(
        wrapper
          .find('PageTitle[title=\'Tudalen Preifatrwydd - GOV.UK\']')
          .exists()
      ).toBeTruthy();
    });

    it('should  display Accept and continue button when privacy has NOT been accepted', () => {
      const englishWrapper = getWrapper('en_UK');
      const welshWrapper = getWrapper('cy_UK');
      expect(englishWrapper.find('#privacyNotice').exists()).toBeTruthy();
      expect(welshWrapper.find('#privacyNotice').exists()).toBeTruthy();
    });

    it('should take the snapshot of the page in English', () => {
      const wrapper = getWrapper('en_UK');
      const { container } = render(wrapper);
      expect(container).toMatchSnapshot();
    });

    it('should handle a submit event', () => {
      const wrapper = getWrapper('en_UK');

      expect(
        wrapper.find('form').simulate('submit', { preventDefault() {} })
      ).toBeTruthy();
    });

    describe('when user has Accepted privacy', () => {
      let wrapper;
      const mockStore = configureStore([thunk]);
      const goBack = jest.fn();

      const getWrapper = (selectedLanguage) => {
        const store = mockStore({
          userAttributes: [
            { name: 'language', value: selectedLanguage },
            { name: 'privacy_statement', value: true },
          ],
        });

        const props = {
          route: {
            history: {
              goBack: goBack(),
            },
            alwaysShowPrivacyPage: true
          },
        };

        window.scrollTo = jest.fn();

        wrapper = mount(
          <Provider store={store}>
            <MemoryRouter>
              <PrivacyStatementPage.default.component {...props} />
            </MemoryRouter>
          </Provider>
        );

        return wrapper;
      };

      it('should load successfully', () => {
        const wrapper = getWrapper();
        const englishWrapper = getWrapper('en_UK');
        const welshWrapper = getWrapper('cy_UK');

        expect(wrapper).toBeDefined();
        expect(englishWrapper).toBeDefined();
        expect(welshWrapper).toBeDefined();
      });

      it('should render the page successfully when the selected language is English', () => {
        const wrapper = getWrapper('en_UK');
        expect(wrapper).toBeDefined();
        expect(
          wrapper.find('PageTitle[title=\'Privacy - GOV.UK\']').exists()
        ).toBeTruthy();
      });
  
      it('should render the page successfully when the selected language is Welsh', () => {
        const wrapper = getWrapper('cy_UK');
        expect(wrapper).toBeDefined();
        expect(
          wrapper
            .find('PageTitle[title=\'Tudalen Preifatrwydd - GOV.UK\']')
            .exists()
        ).toBeTruthy();
      });

      it('should NOT display Accept and continue button when privacy has been accepted', () => {
        const wrapper = getWrapper();
        const englishWrapper = getWrapper('en_UK');
        const welshWrapper = getWrapper('cy_UK');

        expect(wrapper.find('#privacyNotice').exists()).toBeFalsy();
        expect(englishWrapper.find('#privacyNotice').exists()).toBeFalsy();
        expect(welshWrapper.find('#privacyNotice').exists()).toBeFalsy();
      });

      it('redirects to previous page when back button is clicked on English and Welsh page', () => {
        const englishWrapper = getWrapper('en_UK');
        const welshWrapper = getWrapper('cy_UK');
        englishWrapper.find('#back').at(0).simulate('click');
        welshWrapper.find('#back').at(0).simulate('click');
        expect(goBack).toHaveBeenCalled();
      });

      it('should render Redirect component if alwaysShowPrivacyPage is not set to true', () => {
        const store = mockStore({
          userAttributes: [
            { name: 'language', value: 'en_UK' },
            { name: 'privacy_statement', value: true },
          ],
        });
  
        const props = {
          route: {
            history: {
              goBack: goBack(),
            },
            alwaysShowPrivacyPage: false
          },
        };
  
        const wrapper = mount(
          <Provider store={store}>
            <MemoryRouter>
              <PrivacyStatementPage.default.component {...props} />
            </MemoryRouter>
          </Provider>
        );
  
        expect(wrapper).toBeDefined();
        expect(wrapper.find('Redirect').exists()).toBeTruthy();
      });
    });
  });

  it('should call getAllUserAttributes action to load the component', () => {
    const store = { dispatch: jest.fn() };

    getAllUserAttributes.mockReturnValue({ type: 'GET_ALL_USER_ATTRIBUTES' });
  
    PrivacyStatementPageWrapper.loadData(store);

    expect(getAllUserAttributes).toHaveBeenCalled();
  });
});
