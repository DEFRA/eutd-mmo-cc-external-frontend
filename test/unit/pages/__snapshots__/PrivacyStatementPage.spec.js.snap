// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Privacy page when user has not Accepted privacy should take the snapshot of the page in English 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            Privacy notice
          </h1>
          <p>
            Your privacy is extremely important to us. This notice explains what personal information we have, how we use it and how you can check and update any of your personal information. For the purposes of this Privacy Notice, “we” means Department for Environment, Food and Rural Affairs (Defra) and other government agencies in the group.
          </p>
          <p>
             
          </p>
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Why We Collect Your Personal Information
          </h2>
          <p>
            We collect your personal information so you can be issued a UK Export Catch Certificate, by accessing the service operated by Defra and processed by the Marine Management Organisation (MMO) and other UK Marine Fisheries authorities operating in the UK.
          </p>
          <p>
            The following personal information we require is as follows:
          </p>
          <ol
            class="e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Your full name
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Company name (where applicable)
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Address
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Contact telephone number
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Vessel registration details, including:
            </li>
            <ol
              class="e1vcssft0 css-58tcge-StyledList-StyledHoc e4u1u2g0"
            >
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                Vessel name
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                Registration number
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                Port and number
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                International Maritime Organisation (IMO) number
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                Lloyds number
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                Fishing license number
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                Inmarsat number / fax / telephone number where applicable
              </li>
            </ol>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Name of Master of fishing vessel
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Contact details (name, addresses & transportation details) of transshipment, (re)exporter, importer, transporter, processors, and storage facilities.
            </li>
          </ol>
          <br />
          <p>
            We also use your personal information with Google Analytics to monitor your online progress through the service, helping us to improve your experience e.g.
          </p>
          <p>
            Performance Statistics:
          </p>
          <ol
            class="e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Your interaction with the website
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Understanding what a user does when they receive an error (continue or abandon)
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Where you leave the website, which could be from an error page or you may have failed to complete a certificate
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Number of vessels per document
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              How many users created each export certificate
            </li>
          </ol>
          <br />
          <p>
            Performance information:
          </p>
          <ol
            class="e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Catch Certificate Insights
            </li>
            <ol
              class="e1vcssft0 css-58tcge-StyledList-StyledHoc e4u1u2g0"
            >
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                How many catch landing details in each certificate
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                How many variations of products
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                The different transport options people use 
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                The amount of users who change the exporter details
              </li>
            </ol>
            <p>
               
            </p>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Company Insights
            </li>
            <ol
              class="e1vcssft0 css-58tcge-StyledList-StyledHoc e4u1u2g0"
            >
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                The different types of companies
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                The location of companies
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                The amount of documents each company creates
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                The percentage of documents created across all companies
              </li>
              <li
                class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              >
                The amount of companies who create multiple documents
              </li>
            </ol>
          </ol>
          <br />
          <p>
            We do not track IP address or individual User Id’s in Google Analytics.
          </p>
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Who Collects Your Personal Information
          </h2>
          <p>
            The Department for Environment, Food and Rural Affairs (Defra) collects your personal information and is the data controller for the personal information you provide.
          </p>
          <span
            style="font-weight: bold;"
          >
            DEFRA Data Protection Officer
          </span>
           
          <br />
          <br />
          You can contact the Defra Data Protection Officer at:
          <p>
            Department for Environment, 
            <br />
            Food and Rural Affairs
            <br />
            SW Quarter
            <br />
            2nd floor
            <br />
            Seacole Block
            <br />
            2 Marsham Street
            <br />
            London SW1P 4DF 
            <br />
            Email:
             
            <a
              href="mailto:data.protection@defra.gov.uk"
            >
              data.protection@defra.gov.uk
            </a>
          </p>
          <p>
             
          </p>
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Legal Basis for Processing Your Personal Information
             
          </h2>
          <p>
            The legal basis for processing your personal information is in the exercise of official authority, and also to perform specific tasks in the public interest which allow for safe border data transfer. Where required by law, data may be published on UK Government or EU websites.
          </p>
          <p>
            Your personal information will be processed securely using appropriate technical and organisational measures.
          </p>
          <p>
            More information about the MMO can be found at 
            <br />
            <a
              aria-label="Opens link for information about MMO"
              href="https://www.gov.uk/government/organisations/marine-management-organisation"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://www.gov.uk/government/organisations/marine-management-organisation
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
          </p>
          <br />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Who Your Personal Information May be Shared with
          </h2>
          <p>
            Personal information may be made available to public and local authorities and other public bodies in the UK and EU to meet legal requirements in pursuance of public tasks
          </p>
          <p>
            We may have to release information (including personal information and commercial information) under the following legislation:
          </p>
          <ol
            class="e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              General Data Protection Regulation (GDPR)
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              UK Data Protection Act 2018
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Freedom of Information Act 2000
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Environmental Information Regulations 2004
            </li>
          </ol>
          <br />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            How Long We Keep Your Personal Information for
          </h2>
          <p>
            All personal information within Defra is held in accordance with our data retention policy and in alignment with EU Data Privacy statement provisions.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            What will happen if I don’t provide the data?
          </h2>
          <p>
            For statutory and public interest purposes it is necessary to provide data as stated in the relevant legislation.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Will my personal information be transferred outside of the UK / EEA? If it is, how will it be protected?
          </h2>
          <p>
            The data you provide will not be transferred outside of the UK / EEA. However, when necessary personal information may be transferred when it is lawful and complementary to our work carried out in the public interest, i.e. enforcement, verification of documentation outside (UK / EEA),
             
          </p>
          <br />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Your Rights
          </h2>
          <span
            style="font-weight: bold;"
          >
            Your Right of Access
          </span>
           
          <br />
          <br />
          <p>
            You have the right to ask us for copies of your personal information. This right always applies. There are some exemptions, which means you may not always receive all the information we process. You can read more about this right
             
            <a
              aria-label="Opens link about your right of access"
              href="https://ico.org.uk/your-data-matters/your-right-to-get-copies-of-your-data/"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
             
            .
          </p>
          <br />
          <span
            style="font-weight: bold;"
          >
            Your Right to Rectification
          </span>
           
          <br />
          <br />
          <p>
            You have the right to ask us to rectify information you think is inaccurate. You also have the right to ask us to complete information you think is incomplete. This right always applies.You can read more about this right
             
            <a
              aria-label="Opens link about Your right to get your data corrected"
              href="https://ico.org.uk/your-data-matters/your-right-to-get-your-data-corrected/"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
            .
          </p>
          <br />
          <span
            style="font-weight: bold;"
          >
            Your Right to Erasure
          </span>
           
          <br />
          <br />
          <p>
            You have the right to ask us to erase your personal information in certain circumstances. You can read more about this right
             
            <a
              aria-label="Opens link about your right to get your data deleted"
              href="https://ico.org.uk/your-data-matters/your-right-to-get-your-data-deleted/"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
            .
          </p>
          <br />
          <span
            style="font-weight: bold;"
          >
            Your Right to Restriction of Processing
          </span>
           
          <br />
          <br />
          <p>
            You have the right to ask us to restrict the processing of your information in certain circumstances. You can read more about this right
             
            <a
              aria-label="Opens link about Your right to limit how organisations use your data"
              href="https://ico.org.uk/your-data-matters/your-right-to-limit-how-organisations-use-your-data/"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
            .
          </p>
          <br />
          <span
            style="font-weight: bold;"
          >
            Your Right to Object to Processing
          </span>
           
          <br />
          <br />
          <p>
            You have the right to object to processing if we are able to process your information because the process forms part of our public tasks or is in our legitimate interests. You can read more about this right
             
            <a
              aria-label="Opens link about your right to object to the use of your data"
              href="https://ico.org.uk/your-data-matters/the-right-to-object-to-the-use-of-your-data/"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
            .
          </p>
          <br />
          <span
            style="font-weight: bold;"
          >
            Your Right to Data Portability 
          </span>
           
          <br />
          <br />
          <p>
            This only applies to information you have given us. You have the right to ask that we transfer the information you gave us from one organisation to another or give it to you. The right only applies if we are processing information based on your consent or under, or in talks about entering into a contract and the processing is automated. You can read more about this right
             
            <a
              aria-label="Opens link about your right to data portability"
              href="https://ico.org.uk/your-data-matters/your-right-to-data-portability/"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
              <span
                class="govuk-visually-hidden"
              >
                (opens in new tab)
              </span>
            </a>
            .
          </p>
          <p>
            If we are processing your information for criminal law enforcement purposes, your rights are slightly different.
             
          </p>
          <p>
            You are not required to pay any charge for exercising your rights. We have one month to respond to you.
          </p>
          <p>
            Please contact our Data Protection Officer if you wish to make a request
          </p>
          <p>
             
          </p>
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            How do I complain?
          </h2>
          <p>
            You have the right to lodge a complaint with the Independent Commissioners Office (ICO) at any time
          </p>
          <p>
            During the coronavirus pandemic you can still report concerns to the ICO about our information rights practices via their helpline on 0303 123 1113.
          </p>
          <p>
             
          </p>
          <br />
          <form
            action="/orchestration/api/v1/userAttributes"
            method="POST"
          >
            <input
              name="key"
              type="hidden"
              value="privacy_statement"
            />
            <input
              name="value"
              type="hidden"
              value="true"
            />
            <input
              name="nextUri"
              type="hidden"
              value=""
            />
            <button
              class="e1vcssft0 css-1enchvi-StyledButton-StyledHoc e4u1u2g0"
              id="privacyNotice"
              name="privacyNotice"
            >
              Accept and continue
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
